### PR TITLE
parallel-workload: Integrate SQLsmith action

### PR DESCRIPTION
**Pull request by def-**
### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: test/sqlsmith/Dockerfile [hunk 1]
```diff
@@ -28,6 +28,7 @@ ADD https://api.github.com/repos/MaterializeInc/sqlsmith/git/refs/heads/master v
 # Build SQLsmith
 RUN git clone --depth=1 --single-branch --branch=master https://github.com/MaterializeInc/sqlsmith \
     && cd sqlsmith \
+    && git checkout efba94d292c7f32c46e2637a3e05d6f95acf944b \
```
Comment (nrainer-materialize) [inline test/sqlsmith/Dockerfile:31]: Optional: Shall we use a tag?

**Review by nrainer-materialize — APPROVED** [review]

File: test/sqlsmith/Dockerfile (hunk 1 above)
Comment (def-) [reply test/sqlsmith/Dockerfile:31]: I think no real need for versions yet, but it's good to have a fixed SQLsmith version in a version of Materialize code, especially for reproducing issues.

File: misc/python/materialize/parallel_workload/action.py [hunk 2]
```diff
@@ -1286,6 +1355,7 @@ def __init__(
 
 ddl_action_list = ActionList(
     [
+        (SleepAction, 100),  # DDLs easily starve Mz, don't run them too often
```
Comment (philip-stoev) [inline misc/python/materialize/parallel_workload/action.py:1358]: Will this cause too many SQL connections to be actually sleeping at any given time, vs. running DMLs?

You may wish to tackle the imbalance not via `SleepAction`, but by changing the ratio each of the `_list`s is being picked.

File: misc/python/materialize/parallel_workload/database.py [hunk 3]
```diff
@@ -860,3 +863,28 @@ def create(self, exe: Executor) -> None:
 
         for relation in self:
             relation.create(exe)
+
+        result = composition.run(
```
Comment (philip-stoev) [inline misc/python/materialize/parallel_workload/database.py:867]: How much time do you think this action is going to take? Does it make sense, in order to obtain fair scheduling, to limit it to e.g. 1 second or 10 seconds?

**Review by philip-stoev — APPROVED** [review]

File: misc/python/materialize/parallel_workload/action.py (hunk 2 above)
Comment (def-) [reply misc/python/materialize/parallel_workload/action.py:1358]: No, each connection already only runs DMLs or DDLs, never both. This affects only the DDL connections. I noticed that they are kind of drowing out the DML and read connections.

File: misc/python/materialize/parallel_workload/database.py (hunk 3 above)
Comment (def-) [reply misc/python/materialize/parallel_workload/database.py:867]: The run here takes a few seconds, but it only runs once in the beginning to get the catalog state from Materialize.

The interesting part is the SQLsmithAction, which takes less than a second. All it does is get the queries back from SQLsmith. Most of the time was previously spent in getting the catalog state from the database, we only do that once. Only the tables have to be refetched, which is pretty fast. If it turns out to be too slow, can doo

**Pull request merged**